libraries/Wire/src/Wire.cpp

```diff
@@ -241,8 +255,9 @@ size_t TwoWire::write(uint8_t data)
 {
   if(transmitting){
   // in master transmitter mode
-    // don't bother if buffer is full
-    if(txBufferLength >= BUFFER_LENGTH){
+    txBuffer = allocateBuffer(txBuffer, txBufferLength + 1);
```

Comment:
maybe not efficient to allocate +1 byte each time - maybe you could go by steps (allocateBuffer(txBuffer, txBufferLength + BUFFER_LENGTH) each time it is needed) because reallocation can be costly in performances? 
In that way BUFFER_LENGHT would be used as the "unit" of buffer allocation.

Reply:
Right not efficient that's why I've proposed an enhancement in this commit:
https://github.com/fpistm/Arduino_Core_STM32/commit/c4b9f246ed9bf50839117087c480bfe60f11f4f6
Anyway I will add the allocation / BUFFER_LENGTH